src/qiboml/models/decoding.py

```diff
@@ -378,6 +379,47 @@ def output_shape(self) -> tuple[int, int]:
     def analytic(self) -> bool:  # pragma: no cover
         return False
 
+    
+@dataclass(kw_only=True)
+class VariationalQuantumLinearSolver(QuantumDecoding):
+    """Decoder for the Variational Quantum Linear Solver (VQLS).
+    Adapted from the following paper arXiv:1909.05820v4 by Carlos Bravo-Prieto et al. 
+
+    Args:
+        target_state (ndarray): Target solution vector :math:`\\ket{b}`.
+        A (ndarray): The matrix ``A`` in the linear system :math:`A \\, \\ket{x} = \\ket{b}`.
+    
+    Ensure QiboML backend set to Pytorch. 
```

Comment:
```suggestion
    Adapted from the following paper arXiv:1909.05820v4 by Carlos Bravo-Prieto et al. 

    .. note::
        Currently, only implemented for the ``pytorch`` backend.

    Args:
        target_state (ndarray): Target solution vector :math:`\\ket{b}`.
        A (ndarray): The matrix ``A`` in the linear system :math:`A \\, \\ket{x} = \\ket{b}`.
```

Reply:
Also proper article referencing like in #122

Reply:
Yes thanks! 
Think that, here in Qiboml, every decoder is supposed to be used within an optimization pipeline. Do not worry about it, because we have mechanisms to retrieve the gradients implemented in the package. If you are interested, you can have a look into `qiboml.interfaces` :)